meroxa/resource_connector.go

```diff
@@ -283,3 +295,47 @@ func resourceConnectorConfig(d *schema.ResourceData) map[string]interface{} {
 
 	return config
 }
+
+func validateConnectorName() schema.SchemaValidateDiagFunc {
```

Comment:
This function is pretty much copied over [from here](https://github.com/meroxa/platform-api/blob/947d72432b519b43c772dd6d4ea5196333373267/mediator/connector/shared.go#L18-L43). I'd like to consolidate this to a shared library of sorts. Thoughts?

Reply:
Also - thoughts on using `schema.SchemaValidateDiagFunc` here? It seemed like a more clear error type